language/evm/move-to-yul/src/context.rs

```diff
@@ -282,6 +282,15 @@ impl<'a> Context<'a> {
         attributes::is_evm_arith_module(&struct_env.module_env) && struct_env.is_native()
     }
 
+    /// Returns whether the struct identified by module_id and struct_id is the native Table struct.
+    pub fn is_table(&self, struct_id: QualifiedId<StructId>) -> bool {
+        let struct_env = self.env.get_struct(struct_id);
+        format!(
+            "{}",
+            struct_env.get_name().display(struct_env.symbol_pool())
+        ) == "Table"
+    }
```

Comment:
nit: pool.string(struct.env.get_name()).as_ref() == "Table"

However, this should definitely also check for module name and address because currently you match any type with simple name Table.